### ISSUE TITLE
Add user authentication

### ISSUE THREAD

**Issue by awilkinsx** [pelham-wilkins/pw-titan#2]
Build user authentication that checks if LDAP is setup, otherwise utilizes built in local authentication.

**Comment by pelhamrj** [comment]
We can add a Boolean setting to the administration page that allows the end-user to select LDAP or Local Authentication. If LDAP is selected, you can utilize it to authenticate users.

**Comment by awilkinsx** [comment]
Do you think this would be optimal vs just testing if LDAP setting exists?

**Comment by pelhamrj** [comment]
In the event that a URL is changed, a server is moved, etc, I want the end user to be able to reconnect with minimal hassle. We can utilize the .env for the server connection if you prefer, however we need an easy way to access the connection criteria.

**Comment by awilkinsx** [comment]
Keep in mind the end user will not be able to login if the URL is changed or a server moved which results in the loss of an LDAP connection even with a "switch" option. There will be no password saved in the database if they initially used LDAP.

**Comment by awilkinsx** [comment]
Only work around would to force a "password rest" if there is ever an instance where it goes from LDAP to Local auth.

**Comment by pelhamrj** [comment]
This is true. Is there any consideration to self hosting this software in the future rather than giving source to end users? If so, we have more leverage over password resets. Otherwise, the .env file would be good so that a system engineer can modify the environmental variable with the new connection string.
